desrosj/wordpress-props
# Code refactoring and action updates for correct functionality

### Description of the Change
This PR initiates some initial code refactoring and updates the code to ensure that the action functions correctly.

@desrosj Please let me if you need any further information related to this PR.

cc: @jeffpaul

### How to test the Change
1. Set up this action on the specific repository (`iamdharmesh/wordpress-props@main` to test this PR). An example workflow file can be found in the README.md file.
1. Create an issue on the repository.
1. Raise a sample PR that fixes that issue.
1. Verify that the action adds a comment on the PR with a list of contributors.

I have tested this action in one of my test repos and it seems working fine. 
PR link: https://github.com/iamdharmesh/test-actions/pull/12

### Known Caveats/Issues
1. The contributor list only contains users who have an account on wp.org; other users are not included in the comment at the moment.
2. We are collecting contributors from the first 100 PR commits, first 100 comments, first 100 issues comments, first 100 reviews, etc. So, if a PR has more than 100 comments, then contributors who have commented after the first 100 comments will be skipped by this action.
